Backend/Tournament_User/TournamentUserMVSC/controllers/tournamentUserController.js

```diff
@@ -0,0 +1,329 @@
+        console.error('Error retrieving tournaments:', err);
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+/**
+ * Get tournament matchups for a user
+ * @async
+ * @param {Object} req - The request object.
+ * @param {Object} req.params - The request parameters
+ * @param {string} req.params.UUID - The ID of the User.
+ * @param {string} req.params.tournamentId - The ID of the tournament.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Json}} Returns all matchups of the specified tournament that involve the tournament user.Shows player1, player2,playerwon,tournamentID,round number.
+ * @throws {error} If UUID invalid.
+ */
+
+// Get tournament matchups by tournament ID
+exports.getTournamentMatchups = async (req, res, next) => {
+    const { tournamentId } = req.params;
+    const {UUID} = req.body;
+
+    try {
+        const matchups = await TournamentUserService.getTournamentMatchups(tournamentId,UUID);
+        //If no matchups found return error message.
+        if (!matchups || matchups.length === 0) {
+            const error = new Error('No matchups found for the provided tournament ID/UUID');
+            error.statusCode = 404;
+            return next(error);
+        }
+        //Matchup Found
+        res.status(200).json({
+            message: 'Tournament Matchups:',
+            success: true,
+            content: matchups
+        });
+    } catch (err) {
+        console.error('Error fetching matchups:', err);
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+/**
+ * Sign up for a tournament
+ * @async
+ * @param {Object} req - The request object.
+ * @param {Object} req.params - The request parameters
+ * @param {string} req.params.UUID - The ID of the User.
+ * @param {string} req.params.tournamentId - The ID of the tournament.
+ * @param {int} req.params.elo - The elo of the user.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Json} Return User's input as well as success message.
+ * @throws {error} If input invalid or missing fields.
+ */
+
+exports.signUpForTournament = async (req, res, next) => {
+    const { UUID } = req.params;
+    const { tournamentID, elo } = req.body;
+
+    if (!UUID || !tournamentID || !elo) {
+        const error = new Error('Missing required fields (UUID, tournamentID, or elo)');
+        error.statusCode = 400;
+        return next(error);  // Pass error to middleware
+    }
+
+    try {
+        await TournamentUserService.signUpForTournament(UUID, tournamentID, elo);
+        res.status(201).json({
+            message: 'Successfully signed up for the tournament',
+            success: true,
+            content: { UUID, tournamentID, elo }
+        });
+    } catch (err) {
+        if (err.sqlState === '45000') {
+            return res.status(400).send(err.message);  // Custom error from SIGNAL
```

Comment:
actl dont do this can consoleerror and log the error but ideally shouldnt throw the whole database error msg back to the user cos it reveals your database structure